steppyngstounes/parsimoniousStepper.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import division
+from __future__ import unicode_literals
+
+import numpy as np
+from scipy.interpolate import interp1d
+
+from steppyngstounes.stepper import Stepper
+
+__all__ = ["ParsimoniousStepper"]
+
+
+class ParsimoniousStepper(Stepper):
+    r"""Non-monotonic stepper that attempts to find something "interesting".
+
+    Compute the function where the curvature is highest and where not many
+    points have been computed.
+
+    Based on an email::
+
+        From: David Huard <david.huard@gmail.com>
+        Date: April 1, 2009 4:23:14 PM EDT
+        To: "Guyer, Jonathan E. Dr." <jonathan.guyer@nist.gov>
+        Subject: Re: [SciPy-user] "smooth" plots
+
+        Jonathan,
+
+        Here are bits of code I've extracted from a project of mine.
+        You'll have to do some imports from numpy to get it to work.  It's
+        not pretty, but it was enough for my use case.  I'd be interested
+        to look at papers that discuss this problem if you have references
+        nearby.  Note that I also have a version that calls func on
+        multiple processes (using ipython).  With 2 quad cores, this sped
+        up considerably my application.
+
+        HTH,
+
+        David
```

Comment:
```suggestion
    r"""Non-monotonic stepper that attempts to explore sparsely explored regions, with a preference for regions of high curvature.

```

Reply:
Much obliged, David. I hadn't intended to leave the email in, but included it for provenance during development.